wdl-cli/src/inputs/file.rs

```diff
@@ -0,0 +1,111 @@
+//! Input files parsed in from the command line.
+
+use std::path::Path;
+use std::path::PathBuf;
+
+use thiserror::Error;
+use wdl_engine::CompoundValue;
+use wdl_engine::Object;
+use wdl_engine::Value;
+
+use crate::Inputs;
+
+/// An error related to a input file.
+#[derive(Error, Debug)]
+pub enum Error {
+    /// An input file cannot be read from a directory.
+    #[error("an input file cannot be read from the directory: `{0}`")]
+    InvalidDir(PathBuf),
+
+    /// An I/O error.
+    #[error(transparent)]
+    Io(std::io::Error),
+
+    /// The input file did not contain a map at the root.
+    #[error("input file `{0}` did not contain a map at the root")]
+    NonMapRoot(PathBuf),
+
+    /// Neither JSON nor YAML could be parsed from the provided path.
+    #[error("file `{0}` is neither a valid JSON nor a valid YAML file")]
+    UnsupportedFormat(PathBuf),
```

Comment:
Make this more general.

Reply:
On second thought, nevermind. I think it's not likely we will support other input file types any time soon.